### PR TITLE
[feature-wip](statistics) update cache when analysis job finished

### PR DESCRIPTION
**Pull request by Kikyou1997**
# Proposed changes

Issue Number: noissue

## Problem summary

1. Update cache when analysis job finished
2. Rename `StatisticsStorageInitializer` to `InernalSchemaInitializer`

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
4. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
5. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
6. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
7. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.16 seconds
 load time: 442 seconds
 storage size: 17150854212 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221117125724_clickbench_pr_48004.html

File: fe/fe-core/src/main/java/org/apache/doris/catalog/InternalSchemaInitializer.java [hunk 1]
```diff
@@ -46,20 +44,26 @@
...
 
+    /**
+     * If internal table creation failed, will retry after below seconds.
+     */
+    public static final int TABLE_CREATION_RETRY_INTERVAL_IN_SECONDS = 5;
+
+
     public void run() {
-        if (forTest) {
+        if (forTest || (!Env.getCurrentEnv().isMaster() && Env.getCurrentEnv().isReady())) {
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/InternalSchemaInitializer.java:60]: `forTest` means for ut or regression？

Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/InternalSchemaInitializer.java:60]: how about if it is master, but its env is not ready?

File: fe/fe-core/src/main/java/org/apache/doris/statistics/ColumnStatistic.java [hunk 2]
```diff
@@ -110,6 +110,8 @@ public static ColumnStatistic fromResultRow(ResultRow resultRow) {
             columnStatisticBuilder.setNumNulls(Double.parseDouble(resultRow.getColumnValue("null_count")));
             columnStatisticBuilder.setDataSize(Double
                     .parseDouble(resultRow.getColumnValue("data_size_in_bytes")));
+            columnStatisticBuilder.setAvgSizeByte(columnStatisticBuilder.getDataSize()
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/ColumnStatistic.java:113]: if there are some rows to be deleted or rows need to be merged, the AvgSizeByte may becomes larger, is that right?

File: fe/fe-core/src/main/java/org/apache/doris/catalog/InternalSchemaInitializer.java (hunk 1 above)
Comment (Kikyou1997) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/InternalSchemaInitializer.java]: > `forTest` means for ut or regression？

ut

**Review by morrySnow — DISMISSED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**